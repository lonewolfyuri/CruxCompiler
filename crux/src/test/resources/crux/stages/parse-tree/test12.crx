// test if-else

func main() : void {
  if a < b { ::nothing(); }
  if (a >= b) { ::nothing(); }

  if ((((a and b)))) {
    ::nothing();
  } else {
    ::nothing();
  }
  ::done();
}
