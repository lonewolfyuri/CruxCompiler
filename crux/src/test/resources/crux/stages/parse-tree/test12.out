program
  declarationList
    declaration
      functionDefinition
        parameterList
        type
        statementBlock
          statementList
            statement
              ifStatement
                expression0
                  expression1
                    expression2
                      expression3
                        designator
                  op0
                  expression1
                    expression2
                      expression3
                        designator
                statementBlock
                  statementList
                    statement
                      callStatement
                        callExpression
                          expressionList
            statement
              ifStatement
                expression0
                  expression1
                    expression2
                      expression3
                        expression0
                          expression1
                            expression2
                              expression3
                                designator
                          op0
                          expression1
                            expression2
                              expression3
                                designator
                statementBlock
                  statementList
                    statement
                      callStatement
                        callExpression
                          expressionList
            statement
              ifStatement
                expression0
                  expression1
                    expression2
                      expression3
                        expression0
                          expression1
                            expression2
                              expression3
                                expression0
                                  expression1
                                    expression2
                                      expression3
                                        expression0
                                          expression1
                                            expression2
                                              expression3
                                                expression0
                                                  expression1
                                                    expression2
                                                      expression3
                                                        designator
                                                      op2
                                                      expression3
                                                        designator
                statementBlock
                  statementList
                    statement
                      callStatement
                        callExpression
                          expressionList
                statementBlock
                  statementList
                    statement
                      callStatement
                        callExpression
                          expressionList
            statement
              callStatement
                callExpression
                  expressionList
